chat-widget/src/components/livechatwidget/common/updateSessionDataForTelemetry.ts

```diff
@@ -26,12 +26,19 @@ const updateSessionDataForTelemetry = async (chatSDK: any, dispatch: Dispatch<IL
 };
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const updateConversationDataForTelemetry = async (chatSDK: any, dispatch: Dispatch<ILiveChatWidgetAction>) => {
+const updateConversationDataForTelemetry = async (chatSDK: any, dispatch: Dispatch<ILiveChatWidgetAction>, maxAttempts = 3, attempt = 1) => {
     if (chatSDK) {
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
         const liveWorkItem: any = await getConversationDetailsCall(chatSDK);
-        const telemetryData = TelemetryHelper.addConversationDataToTelemetry(liveWorkItem, TelemetryManager.InternalTelemetryData);
-        dispatch({ type: LiveChatWidgetActionType.SET_TELEMETRY_DATA, payload: telemetryData });
-        BroadcastService.postMessage({ eventName: BroadcastEvent.UpdateConversationDataForTelemetry, payload: {liveWorkItem}});
+        if (liveWorkItem.conversationId) {
+            const telemetryData = TelemetryHelper.addConversationDataToTelemetry(liveWorkItem, TelemetryManager.InternalTelemetryData);
+            dispatch({ type: LiveChatWidgetActionType.SET_TELEMETRY_DATA, payload: telemetryData });
+            BroadcastService.postMessage({ eventName: BroadcastEvent.UpdateConversationDataForTelemetry, payload: {liveWorkItem}});
+        } else if (attempt < maxAttempts) {
```

Comment:
Don't we have Axios for retries?

Reply:
Axios automatically retries requests for retriable errors, such as when error.response.status is between 500 and 599. In this case, the response indicates 'not found,' meaning the chat doesn't exist. I don't think we need to modify Axios to handle this type of response.